### PR TITLE
added sorting support by object.object.value

### PR DESCRIPTION
**Pull request by Vitaly-B**
added sorting support by object.object.value

### CODE REVIEW

File: src/SortByFieldExtension.php [hunk 1]
```diff
@@ -84,11 +96,27 @@ public function sortByFieldFilter($content, $sort_by = null, $direction = 'asc')
...
-        if (is_array($item))
-            return array_key_exists($field, $item);
-        elseif (is_object($item))
-            return isset($item->$field) || property_exists($item, $field);
-        else
-            return false;
+
+        $isSortable = false;
+
+        foreach (explode('.', $field) as $field) {
```
Comment (victorhaggqvist) [inline src/SortByFieldExtension.php]: Reassigning param `$field`, don't do that

File: src/SortByFieldExtension.php [hunk 2]
```diff
@@ -84,11 +96,27 @@ public function sortByFieldFilter($content, $sort_by = null, $direction = 'asc')
...
+            if (is_array($item)) {
+                $isSortable = array_key_exists($field, $item);
+                if ($isSortable) {
+                    $item = $item[$field];
```
Comment (victorhaggqvist) [inline src/SortByFieldExtension.php]: Reassigning param `$item`, don't do that

File: src/SortByFieldExtension.php [hunk 3]
```diff
@@ -84,11 +96,27 @@ public function sortByFieldFilter($content, $sort_by = null, $direction = 'asc')
...
+                }
+            } elseif (is_object($item)) {
+                $isSortable = isset($item->$field) ||
+                    method_exists($item, 'get'.ucfirst($field));
+                if ($isSortable) {
+                    $item = isset($item->$field)
```
Comment (victorhaggqvist) [inline src/SortByFieldExtension.php]: Nested ternary operator on multiple lines, should properly not be a ternary operator. This is not particularity readable.

**Review by victorhaggqvist — CHANGES_REQUESTED** [review]
While I can see this being useful, I am somewhat on the fence of adding this on under the same extension filter since it adds quite a bit of complexity.

See the comments and add test cases for failing scenarios.

File: src/SortByFieldExtension.php (hunk 3 above)
Comment (Vitaly-B) [reply src/SortByFieldExtension.php]: I`am pushed fixes

**Comment by Vitaly-B** [comment]
Hi, I thought that it could be made easier
     public function getFilters()
    {
        return [
            new \Twig_SimpleFilter('sort_by_field', function ($data, $propertyPath = null, $direction = 'asc') {

                $flip = ($direction === 'desc') ? -1 : 1;

                usort($data, function($a, $b) use($propertyPath, $direction, $flip) {

                    if($propertyPath) {
                        $propertyAccessor = PropertyAccess::createPropertyAccessor();

                        $a_sort_value = $propertyAccessor->getValue($a, $propertyPath);
                        $b_sort_value = $propertyAccessor->getValue($b, $propertyPath);
                    } else {
                        $a_sort_value = $a;
                        $b_sort_value = $b;
                    }

                    if ($a_sort_value == $b_sort_value) {
                        return 0;
                    } else if ($a_sort_value > $b_sort_value) {
                        return (1 * $flip);
                    } else {
                        return (-1 * $flip);
                    }
                });

                return $data;
            })
        ];
    }

**Comment by Vitaly-B** [comment]
...

**Pull request closed without merging**